packages/@aws-cdk/aws-iam/lib/principals.ts

```diff
@@ -382,6 +382,31 @@ export class FederatedPrincipal extends PrincipalBase {
   }
 }
 
+/**
+ * A principal that represents a federated identity provider as Web Identity such as Cognito, Amazon,
+ * Facebook, Google, etc.
+ */
+export class WebIdentityPrincipal extends FederatedPrincipal {
+
+  /**
+   *
+   * @param identityProvider identity provider (i.e. 'cognito-identity.amazonaws.com' for users authenticated through Cognito)
+   * @param conditions The conditions under which the policy is in effect.
+   *   See [the IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition.html).
+   */
+  constructor(identityProvider: string, conditions: Conditions = {}) {
```

Comment:
Would be nice to make this a not a `string` or maybe just add subclasses such as `OpenIdPrincipal` that extends this an accepts an `IOpenIdProvider` object.